sewer/auth.py

```diff
@@ -0,0 +1,77 @@
+import logging
+import base64
+
+
+def calculate_safe_base64(un_encoded_data):
+    """
+    takes in a string or bytes
+    returns a string
+    """
+    if isinstance(un_encoded_data, str):
+        un_encoded_data = un_encoded_data.encode("utf8")
+    r = base64.urlsafe_b64encode(un_encoded_data).rstrip(b"=")
+    return r.decode("utf8")
+
+
+class BaseAuthProvider(object):
+    def __init__(self, auth_type, LOG_LEVEL="INFO"):
+        self.LOG_LEVEL = LOG_LEVEL
+        self.dns_provider_name = self.__class__.__name__
+
+        self.logger = logging.getLogger(__name__)
+        self.logger.setLevel(self.LOG_LEVEL)
+        if not self.logger.hasHandlers():
+            handler = logging.StreamHandler()
+            formatter = logging.Formatter("%(message)s")
+            handler.setFormatter(formatter)
+            self.logger.addHandler(handler)
+
+        self.auth_type = auth_type
+
+    def log_response(self, response):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method could be a function](https://app.codacy.com/manual/komuW/sewer/pullRequest?prid=5008291)